Tasks/DeployAzureResourceGroup/Utility.ps1

```diff
@@ -896,25 +923,28 @@ function Add-AzureVMCustomScriptExtension
     try
     {
         $isExtensionExists = Is-WinRMCustomScriptExtensionExists -resourceGroupName $resourceGroupName -vmName $vmName -extensionName $extensionName
-    Write-Verbose -Verbose "IsExtensionExists: $isExtensionExists"
+        Write-Verbose -Verbose "IsExtensionExists: $isExtensionExists"
 	
-    if($isExtensionExists)
-    {
-        Write-Verbose -Verbose "Skipping the addition of custom script extension '$extensionName' as it already exists"
-        return
-    }
+        if($isExtensionExists)
+        {
+            Write-Verbose -Verbose "Skipping the addition of custom script extension '$extensionName' as it already exists"
+
+            Add-WinRMHttpsNetworkSecurityRuleConfig -resourceGroupName $resourceGroupName -vmId $vmId -ruleName $ruleName -rulePriotity $rulePriotity -winrmHttpsPort $winrmHttpsPort
```

Comment:
If the addition of security rule is failed in the first attempt then, it never adds the security group if we miss here.
This is a quick call which is idempotent.